crates/ty_python_semantic/src/types.rs

```diff
@@ -1170,6 +1170,16 @@ impl<'db> Type<'db> {
                 target.is_equivalent_to(db, Type::object(db))
             }
 
+            (Type::FunctionLiteral(self_function), Type::FunctionLiteral(target_function)) => {
```

Comment:
nit: it might be useful to copy / move the comment present in `FunctionType::is_subtype_of` right above these three branches to clarify the intent here similar to how there are other comments right in these match expression. It might also remove the duplication between the comments in `FunctionType` and `BoundMethodType`

Same comment for other relation methods where the comments are present.

Reply:
Good idea, done.  (I kept some comments in `FunctionType` and `BoundMethodType` which are more specific to those variants, but also put some commentary here about this cluster of match arms that are all about type variants that contain a function literal)